source/FreeRTOS_IPv6.c

```diff
@@ -55,6 +64,63 @@ const struct xIPv6_Address FreeRTOS_in6addr_any = { 0 };
  */
 const struct xIPv6_Address FreeRTOS_in6addr_loopback = { { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1 } };
 
+/**
+ * This variable is initialized by the system to contain the unspecified IPv6 address.
+ */
+static const struct xIPv6_Address xIPv6UnspecifiedAddress = { { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 } };
+
+/*
+ * Check if the packet is a legal loopback packet.
+ */
+static BaseType_t xIsIPv6Loopback( const IPHeader_IPv6_t * const pxIPv6Header );
+
+/**
+ * @brief Get the group ID and stored into IPv6_Address_t.
+ *
+ * @param[in] pxIPv6Address: The multicast address to filter group ID.
+ *
+ * @return IPv6_Address_t with group ID only.
+ */
+static IPv6_Address_t xGetIPv6MulticastGroupID( const IPv6_Address_t * pxIPv6Address )
+{
+    IPv6_Address_t xReturnGroupID = { 0 };
+
+    configASSERT( pxIPv6Address != NULL );
+
+    ( void ) memcpy( &( xReturnGroupID.ucBytes[ 2 ] ), &( pxIPv6Address->ucBytes[ 2 ] ), 14 );
+
+    return xReturnGroupID;
+}
+
+/**
+ * @brief Check if the packet is a legal loopback packet.
+ *
+ * @param[in] pxIPv6Header: The IP packet in pxNetworkBuffer.
+ *
+ * @return Returns pdTRUE if it's a legal loopback packet, pdFALSE if not .
+ */
+/* MISRA Ref 8.9.1 [File scoped variables] */
+/* More details at: https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/main/MISRA.md#rule-89 */
+/* coverity[misra_c_2012_rule_8_9_violation] */
+static BaseType_t xIsIPv6Loopback( const IPHeader_IPv6_t * const pxIPv6Header )
+{
+    BaseType_t xReturn = pdFALSE;
+    const NetworkEndPoint_t * pxEndPoint = FreeRTOS_FindEndPointOnIP_IPv6( &( pxIPv6Header->xSourceAddress ) );
+
+    /* Allow loopback packets from this node itself only. */
+    if( ( pxEndPoint != NULL ) &&
+        ( memcmp( pxIPv6Header->xDestinationAddress.ucBytes, in6addr_loopback.ucBytes, sizeof( IPv6_Address_t ) ) == 0 ) &&
+        ( memcmp( pxIPv6Header->xSourceAddress.ucBytes, pxEndPoint->ipv6_settings.xIPAddress.ucBytes, sizeof( IPv6_Address_t ) ) == 0 ) )
+    {
+        xReturn = pdTRUE;
+    }
+
+    return xReturn;
+}
+
+
+/*-----------------------------------------------------------*/
+
 /**
  * @brief Check whether this IPv6 address is a multicast address or not.
```

Comment:
Please change the description as well to something like "Check whether this IPv6 address is an allowed multicast address or not." And add the description for return value as well.